src/config.zig

```diff
@@ -180,6 +180,35 @@ pub const io_depth_read = 8;
 /// The maximum number of concurrent write I/O operations to allow at once.
 pub const io_depth_write = 8;
 
+/// The number of redundant copies of the superblock in the superblock storage zone.
+/// This must be either { 4, 6, 8 }, i.e. an even number, for more efficient flexible quorums.
+/// This is further multiplied by two to support copy-on-write across copy sets.
+///
+/// The superblock contains local state for the replica and therefore cannot be replicated remotely.
+/// Loss of the superblock would represent loss of the replica and so it must be protected.
+/// Since each superblock copy also copies the superblock trailer (around 33 MiB), setting this
+/// beyond 4 copies (or decreasing block_size < 64 KiB) can result in a superblock zone > 264 MiB.
+///
+/// This can mean checkpointing latencies in the rare extreme worst-case of at most 264ms, although
+/// this would require EWAH compression of our block free set to have zero effective compression.
+/// In practice, checkpointing latency should be an order of magnitude better due to compression,
+/// because our block free set will fill holes when allocating.
+///
+/// The superblock only needs to be checkpointed every now and then, before the WAL wraps around,
+/// or when a view change needs to take place to elect a new primary.
+pub const superblock_copies = 4;
+
+/// The maximum size of a local data file.
+/// This should not be much larger than several TiB to limit:
+/// * blast radius and recovery time when a whole replica is lost,
+/// * replicated storage overhead, since all data files are mirrored,
+/// * the size of the superblock storage zone, and
+/// * the static memory allocation required for tracking LSM forest metadata in memory.
+pub const size_max = 16 * 1024 * 1024 * 1024 * 1024;
```

Comment:
> Would it make sense to use a much lower maximum when deployment_environment != .production?

That's a good question I hadn't considered yet. It might, this would reduce the superblock size from 256 MiB to something smaller. However, at the same time, it would mask latency issues in the development environment which might be good for the developer to feel. Storage is also less constrained in development than memory, which is where we have tended to switch on `deployment_environment` in the past.

> What happens when a replica reaches size_max? Will it reject all future prepare requests?

Yes, similar to hitting `ENOSPC`. `size_max` is our upper bound on the design for how much local storage we can address. Setting this too large would at some point impact the design. 16 TiB is similar to limits on other databases. On a 1 Gbps NIC, it would take a minimum of 36-48 hours to recover the loss of that much data.

> Assert that this (and also block_size) is a multiple of sector_size.

I would like us to have a separate file to do all the asserts and verification of what will become `config_comptime.zig` separately. At present, we do have these sector multiple asserts where the variables are used but we do need to verify these as soon as possible at comptime.